aspnetcore/mvc/models/file-uploads.md

```diff
@@ -1,195 +1,656 @@
+<xref:Microsoft.AspNetCore.Mvc.RequestFormLimitsAttribute> is used to set the <xref:Microsoft.AspNetCore.Http.Features.FormOptions.MultipartBodyLengthLimit> for a single page or action.
 
-The following error indicates your file upload exceeds the server's configured `maxAllowedContentLength`:
+In a Razor Pages app, apply the filter with a [convention](xref:razor-pages/razor-pages-conventions) in `Startup.ConfigureServices`:
 
+```csharp
+services.AddMvc()
+    .AddRazorPagesOptions(options =>
+        {
+            options.Conventions
+                .AddPageApplicationModelConvention("/FileUploadPage",
+                    model.Filters.Add(
+                        new RequestFormLimitsAttribute()
+                        {
+                            // Set the limit to 256 MB
+                            MultipartBodyLengthLimit = 268435456
+                        });
+        })
+    .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
 ```
-HTTP 404.13 - Not Found
-The request filtering module is configured to deny a request that exceeds the request content length.
+
+In a Razor pages app or an MVC app, apply the filter to the page handler class or action method:
+
+```csharp
+// Set the limit to 256 MB
+[RequestFormLimits(MultipartBodyLengthLimit = 268435456)]
+public class BufferedSingleFileUploadPhysicalModel : PageModel
+{
+    ...
+}
 ```
 
-The default setting is `30000000`, which is approximately 28.6MB. The value can be customized by editing *web.config*:
+### Kestrel maximum request body size
+
+For apps hosted by Kestrel, the default maximum request body size is 30,000,000 bytes, which is approximately 28.6 MB. Customize the limit using the [MaxRequestBodySize](xref:fundamentals/servers/kestrel#maximum-request-body-size) Kestrel server option:
+
+::: moniker range=">= aspnetcore-3.0"
+
+```csharp
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseStartup<Startup>();
+        })
+        .ConfigureKestrel((context, options) =>
+        {
+            // Handle requests up to 50 MB
+            options.Limits.MaxRequestBodySize = 52428800;
+        });
+```
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.0"
+
+```csharp
+public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+    WebHost.CreateDefaultBuilder(args)
+        .UseStartup<Startup>()
+        .ConfigureKestrel((context, options) =>
+        {
+            // Handle requests up to 50 MB
+            options.Limits.MaxRequestBodySize = 52428800;
+        });
+```
+
+::: moniker-end
+
+<xref:Microsoft.AspNetCore.Mvc.RequestSizeLimitAttribute> is used to set the [MaxRequestBodySize](xref:fundamentals/servers/kestrel#maximum-request-body-size) for a single page or action.
+
+In a Razor Pages app, apply the filter with a [convention](xref:razor-pages/razor-pages-conventions) in `Startup.ConfigureServices`:
```

Comment:
Super untested, but in 3.0, you should be able to do

`@attribute [RequestSizeLimitAttribute(52428800)]` in the cshtml

Reply:
I'll note this on the 3.0 update issue.